### PR TITLE
Multiple hosts

### PR DESCRIPTION
**Pull request by Jetski5822**
dependent on https://github.com/OrchardCMS/Orchard2/pull/633

### CODE REVIEW

File: src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs [hunk 1]
```diff
@@ -126,9 +133,9 @@ private string GetHostAndPrefix(string host, string appRelativePath)
 
         }
 
-        private string GetHostAndPrefix(ShellSettings shellSettings)
+        private string[] GetHostsWithPrefix(ShellSettings shellSettings)
         {
-            return shellSettings.RequestUrlHost + "/" + shellSettings.RequestUrlPrefix;
+            return shellSettings.RequestUrlHost.Split(',').Select(ruh => ruh + "/" + shellSettings.RequestUrlPrefix).ToArray();
```
Comment (sebastienros) [inline src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs]: add `' '` and use StringSplitOptions.RemoveEmtpy.

File: src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs [hunk 2]
```diff
@@ -126,9 +133,9 @@ private string GetHostAndPrefix(string host, string appRelativePath)
 
         }
 
-        private string GetHostAndPrefix(ShellSettings shellSettings)
+        private string[] GetHostsWithPrefix(ShellSettings shellSettings)
```
Comment (sebastienros) [inline src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs]: How often is it called per request per app lifetime? Concern is perf. If only called once then GTG.

**Comment by sebastienros** [comment]
Have you tried it works?

File: src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs (hunk 2 above)
Comment (Jetski5822) [inline src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs]: Good point. I'll check.

File: src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs (hunk 1 above)
Comment (Jetski5822) [inline src/OrchardCore/Orchard.Environment.Shell/RunningShellTable.cs]: Done

**Comment by sebastienros** [comment]
Looks a lot like the code in O1 at first sight, right?

**Comment by Jetski5822** [comment]
For now, green tests more important at the moment. Slowly refactoring :)

**Comment by sebastienros** [comment]
I will look at the tests (thanks) to see what scenarios you support and see if there is a simpler way to do it.
I intentionally moved away from the code in O1.

**Comment by Jetski5822** [comment]
I know you did... but as the O1 stuff worked with what I needed and supports both the *. and ',' scenarios... it was easier to move back, put the tests in, then refactor.

**Comment by Jetski5822** [comment]
Oh these tests are O1 tests... just ported to XUnit.

**Pull request merged**